src/gps/gps/rtcm_pub_node.py

```diff
@@ -0,0 +1,239 @@
+import threading
+import rclpy
+import serial
+from rclpy.node import Node
+from rtcm_msgs.msg import Message as Rtcm
+from pyubx2.ubxtypes_configdb import SET_LAYER_RAM, SET_LAYER_BBR, SET_LAYER_FLASH
+from pyubx2 import (
+    RTCM3_PROTOCOL,
+    UBXMessage,
+    UBXMessageError,
+    UBXParseError,
+    UBXReader,
+    VALCKSUM,
+)
+
+# Defines
+TMODE_SVIN = 1
+TMODE_FIXED = 2
+
+
+class IoManager:
+    """
+    Manages serial I/O operations, ensuring thread safety for reading and writing data.
+
+    Attributes:
+        lock (threading.Lock): Ensures thread-safe access to the serial port.
+        worker (serial.Serial): Serial connection instance.
+        ubr (UBXReader): UBXReader instance for parsing RTCM/UBX data.
+    """
+
+    def __init__(self, port="/dev/ttyACM0", baud=38400):
+        """
+        Initializes the serial connection and UBXReader.
+
+        Args:
+            port (str): Serial port to connect to.
+            baud (int): Baud rate for the serial connection.
+
+        Raises:
+            RuntimeError: If the serial port cannot be opened.
+        """
+        self.lock = threading.Lock()
+        with self.lock:
+            try:
+                self.worker = serial.Serial(port, baud, timeout=1)
```

Comment:
Magic number could be replaced with an informative constant or even better add it to the constructor to the manager class (with a default value) to allow for changes to be made more easily.